### ISSUE TITLE
Documentation

### ISSUE THREAD

**Issue by rweigel** [hapi-server/client-java#2]
Can you add to README.md instructions for compiling and testing a simple application that uses the code, e.g.

```bash
git clone https://github.com/hapi-server/client-java.git
cd client-java; javacc demo.java # or ant ...
java -jar demo.jar # Gets a dataset and reports to stdout metadata and numbers
```

You could also provide an example using the pre-compiled jar file.

I'd also put the `WorkingWith*` files in README.md as they are important and useful and the existing README.md will be short.

I'm not sure about putting Netbeans-specific stuff in there. Do the majority of Java developers use Netbeans? Do you need Netbeans to to compile and use? I generally keep my Emacs and Sublime-related project configuration files out of repositories.

Is it more common to have a `bin` rather than `store` directory? Not sure what is used in java world.

**Comment by jbfaden** [comment]
Yes, the .nbproject stuff is just to make my life easier as I develop the code.  Right now there are two .java codes needed, and my plan was to combine them (put the TimeUtil stuff into the URITemplate code) into one .java file.  Making a jar might be a better way, because then it could be made into a Maven class and get exposure there.  I've never published anything using this mechanism, but I've used pre-compiled jar files from Maven (Central) often.

**Comment by jbfaden** [comment]
I got confused about which project we were talking about, I was thinking this was the URI_Templates stuff.  For this Java client for HAPI, I can add instructions for building with ant and javacc.  I'd still like for this to be a Netbeans project until it is mature, and I can add a comment about how the .nbproject can be ignored.
